src/libs/data-utils.js

```diff
@@ -340,3 +355,249 @@ export const EntityUploader = class EntityUploader extends Component {
+        ['string', () => _.trim]
+      )
+      const preparedValue = isList ? _.map(prepFn, newValue) : prepFn(newValue)
+
+      await Ajax()
+        .Workspaces
+        .workspace(namespace, name)
+        .upsertEntities([{ name: entityName, entityType, attributes: { [attributeName]: preparedValue } }])
+      onSuccess()
+    } catch (e) {
+      onDismiss()
+      reportError('Unable to modify entity', e)
+    }
+  }
+
+  const doDelete = async () => {
+    try {
+      setIsBusy(true)
+      await Ajax().Workspaces.workspace(namespace, name).deleteEntityAttribute(entityType, entityName, attributeName)
+      onSuccess()
+    } catch (e) {
+      onDismiss()
+      reportError('Unable to modify entity', e)
+    }
+  }
+
+  const boldish = text => span({ style: { fontWeight: 600 } }, [text])
+
+  const makeInput = ({ value = '', ...props }) => Utils.switchCase(editType,
+    ['number', () => h(NumberInput, { autoFocus: true, placeholder: 'Enter a value', value, ...props })],
+    ['boolean', () => div({
+      style: { flexGrow: 1, display: 'flex', alignItems: 'center', height: '2.25rem' }
+    }, [h(Switch, { checked: value, ...props })])],
+    [Utils.DEFAULT, () => h(TextInput, {
+      autoFocus: true,
+      placeholder: editType === 'reference' ? `Enter a ${linkedEntityType}_id` : 'Enter a value',
+      value,
+      ...props
+    })]
+  )
+
+  return h(Modal, {
+    title: 'Modify Attribute',
+    onDismiss,
+    showButtons: false
+  }, [
+    consideringDelete ?
+      h(Fragment, [
+        'Are you sure you want to delete the attribute ', boldish(attributeName),
+        ' from the ', boldish(entityType), ' called ', boldish(entityName), '?',
+        div({ style: { marginTop: '1rem' } }, [boldish('This cannot be undone.')]),
+        div({ style: { marginTop: '1rem', display: 'flex', alignItems: 'baseline' } }, [
+          div({ style: { flexGrow: 1 } }),
+          h(ButtonSecondary, { style: { marginRight: '1rem' }, onClick: () => setConsideringDelete(false) }, ['Back to editing']),
+          h(ButtonPrimary, { onClick: doDelete }, ['Delete Attribute'])
+        ])
+      ]) :
+      h(Fragment, [
+        div({ style: { marginBottom: '1rem' } }, [
+          fieldset({ style: { border: 'none', margin: 0, padding: 0 } }, [
+            legend({ style: { marginBottom: '0.5rem' } }, [isList ? 'List item type:' : 'Attribute type:']),
+            h(Fragment, _.map(({ type, tooltip }) => h(TooltipTrigger, {
+              content: tooltip
+            }, [
+              span({ style: { marginRight: '1.2rem' } }, [h(RadioButton, {
+                text: _.startCase(type),
+                name: 'edit-type',
+                checked: editType === type,
+                onChange: () => {
+                  const convertFn = type === 'number' ?
+                    v => {
+                      const numberVal = _.toNumber(v)
+                      return _.isNaN(numberVal) ? '' : numberVal
```

Comment:
It looks like the server won't save mixed arrays, so this should either default to `0`, or we should provide some additional validation to avoid trying to save a value that will fail.